wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallController/ActiveCallRouter.swift

```diff
@@ -151,22 +165,56 @@ extension ActiveCallRouter: ActiveCallRouterProtocol {
     }
 
     // MARK: - Alerts
-    func presentSecurityDegradedAlert(for reason: CallDegradationReason) {
-        executeOrSchedulePostCallAction { [weak self] in
+
+    func presentSecurityDegradedAlert(for reason: CallDegradationReason, callEnded: Bool = false, completion: @escaping (AlertChoice) -> Void) {
+        guard self.presentedDegradedAlert == nil else {
+            completion(.alreadyPresented)
+            return
+        }
+
+        executeOrSchedulePostCallAction { [weak self] postCallActionCompletion in
             let alert: UIAlertController
             switch reason {
             case .degradedUser(user: let user):
-                alert = UIAlertController.degradedCall(degradedUser: user?.value, callEnded: true)
+                alert = UIAlertController.degradedCall(degradedUser: user?.value, callEnded: callEnded, confirmationBlock: { continueDegradedCall in
+                    completion(continueDegradedCall ? .confirm : .cancel)
+                    postCallActionCompletion()
+                    self?.presentedDegradedAlert = nil
+                })
             case .invalidCertificate:
-                alert = UIAlertController.degradedMLSConference(conferenceEnded: true)
+                if !callEnded {
+                    alert = UIAlertController.incomingCallDegradedMLSConference(confirmationBlock: { answerDegradedCall in
+                        completion(answerDegradedCall ? .confirm : .cancel)
+                        postCallActionCompletion()
+                        self?.presentedDegradedAlert = nil
+                    })
+                } else {
+                    alert = UIAlertController.degradedMLSConference(conferenceEnded: callEnded, cancelBlock: {
+                        completion(.ok)
+                        postCallActionCompletion()
+                        self?.presentedDegradedAlert = nil
+                    })
+                }
             }
+
+            self?.presentedDegradedAlert = alert
+
             self?.rootViewController.present(alert, animated: true)
         }
     }
 
+    func dismissSecurityDegradedAlertIfNeeded() {
+        guard let alert = self.presentedDegradedAlert else { return }
+
+        alert.dismissIfNeeded()
+        self.presentedDegradedAlert = nil
+    }
```

Comment:
With `presentedDegradedAlert` as `weak` (see other comment) this func can be replaced by one line:
`presentedDegradedAlert?.dismiss(animated: animated, completion: completion)`